### DISCUSSION TITLE
NavLink migration from react-router

### DISCUSSION THREAD

**Discussion by devinrhode2** [TanStack/router#2422] [Q&A]
We are using tanstack router for a new project.

It is a totally separate vite app, on a new subdomain, but the header will be the same as our current platform.

Our current platform uses react-router.

We are going to share code via git submodules, so that we retain seamless hot reloading when the shared code changes. (happy to discuss submodule vs npm more in a different thread)

First challenge we are facing is with NavLink. 

Would anyone be willing to share their implementation of a tanstack NavLink?

**Comment by SeanCassiere** [comment]
TanStack Router has its own `<Link>` component: https://tanstack.com/router/latest/docs/framework/react/api/router/linkComponent
+ Plus a guide on using Custom interfaces as a "link" for navigation: https://tanstack.com/router/latest/docs/framework/react/guide/custom-link

**Reply by devinrhode2** [reply]
how do I take a relative path from the "to" prop and resolve it to the absolute path?

All Im trying to do is re-create the react-router-dom `className={({isActive})=>` callback.

**Reply by devinrhode2** [reply]
in my case i only happen to need to base path

I suppose i could resolve it manually, but im sure theres a util in router already

**Reply by SeanCassiere** [reply]
I'm not sure that I fully understand what your basepath situation is.

In TanStack Router, once you've configured your basepath on the [router instance](https://tanstack.com/router/latest/docs/framework/react/api/router/RouterOptionsType#basepath-property), it then automatically resolves your `to` pathname with the basepath value with what you've configured.
So, if you've configured your basepath as `/app` and set the `to` value to `/dashboard` on the `<Link>` component, then when clicked it gets resolved to `/app/dashboard` in the URL bar.

Edit: But if you need the basepath value (that's not for navigation purposes, i.e. application logic reasons), you can always get it off of `useRouter().basepath`.

---

For styling, you have the `className` attr, as well as options for `activeProps` and `inactiveProps` which also accept the className attr in there. See the example here: https://github.com/TanStack/router/blob/a42deba77679f221d688eef5e6f121d7bc00731b/examples/react/basic-file-based/src/routes/__root.tsx#L30-L37

Edit: Your CSS selectors could also target data-attributes as well.
https://github.com/TanStack/router/blob/a42deba77679f221d688eef5e6f121d7bc00731b/packages/react-router/src/link.tsx#L847-L848
